sdk/core/Azure.Core/src/Serialization/JsonObjectSerializer.cs

```diff
@@ -67,6 +67,20 @@ public override async ValueTask SerializeAsync(Stream stream, object? value, Typ
             return JsonSerializer.DeserializeAsync(stream, returnType, _options, cancellationToken);
         }
 
+        /// <inheritdoc />
+        public override BinaryData Serialize(object? value, Type? inputType = default, CancellationToken cancellationToken = default) =>
+            SerializeToBinaryDataInternal(value, inputType);
+
+        /// <inheritdoc />
+        public override ValueTask<BinaryData> SerializeAsync(object? value, Type? inputType = default, CancellationToken cancellationToken = default) =>
+             new ValueTask<BinaryData>(Task.FromResult(SerializeToBinaryDataInternal(value, inputType)));
+
+        private BinaryData SerializeToBinaryDataInternal(object? value, Type? inputType)
+        {
+            byte[] bytes = JsonSerializer.SerializeToUtf8Bytes(value, inputType ?? value?.GetType() ?? typeof(object), _options);
```

Comment:
The jsonSerializable ctor of BinaryData actually does this, but it seems better to be explicit here in terms of the serializer being used.